include/veriblock/pop/blockchain/blockchain_util.hpp

```diff
@@ -111,13 +111,24 @@ bool recoverEndorsements(ProtectedBlockTree& ed_,
       auto& by = endorsed->getEndorsedBy();
       VBK_ASSERT_MSG(std::find(by.begin(), by.end(), endorsement) == by.end(),
                      "same endorsement is added to endorsedBy second time");
+      bool isDirty = endorsed->isDirty();
       endorsed->insertEndorsedBy(endorsement);
+      // keep dirty flag since recoverEndorsements is used when loading blocks from storage
+      // and should not affect dirtyness of the blocks
+      if (!isDirty) {
+        endorsed->unsetDirty();
+      }
```

Comment:
I think you can just do `endorsed->unsetDirty()` regardless of its value. (so remove bool isDirty and if)

Reply:
We can try it this way but I thought it is better to preserve existing state.